src/main/java/org/opensearch/agent/tools/IndexRoutingTool.java

```diff
@@ -0,0 +1,315 @@
+                + "mappings of index-1\n"
+                + "Sample data:\n"
+                + "data from index-1\n"
+                + "---\n"
+                + "index-2: Index Mappings:\n"
+                + "mappings of index-2\n"
+                + "Sample data:\n"
+                + "data from index-2\n"
+                + "---\n"
+                + "...\n"
+                + "\n"
+                + "Now the actual index mappings and sample data begins:\n"
+                + "{summaries}\n"
+                + "\n"
+                + "-------------------\n"
+                + "\n"
+                + "Format the output as a comma-separated sequence, e.g. index-1, index-2, index-3. If no indexes \n"
+                + "appear relevant to the question, return the empty string ''.\n"
+                + "\n"
+                + "Just return the index names, nothing else. \n"
+                + "If you are not sure, just return 'Not sure', nothing else.\n"
+                + "\n"
+                + "Question: {question}\n"
+                + "Answer:"
+        );
+
+    public static final int DEFAULT_K = 5;
+    public static String EMBEDDING_MODEL_ID = "embedding_model_id";
+    public static String INFERENCE_MODEL_ID = "inference_model_id";
+    public static String PROMPT_TEMPLATE = "prompt_template";
+
+    @Setter
+    @Getter
+    private String name = TYPE;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+
+    @Getter
+    @Setter
+    private String inferenceModelId;
+
+    private final MLClients mlClients;
+
+    @Setter
+    private String promptTemplate;
+
+    public IndexRoutingTool(
+        Client client,
+        NamedXContentRegistry xContentRegistry,
+        Integer docSize,
+        Integer k,
+        String embeddingModelId,
+        String inferenceModelId
+    ) {
+        super(
+            client,
+            xContentRegistry,
+            IndexSummaryEmbeddingJob.INDEX_SUMMARY_EMBEDDING_INDEX,
+            IndexSummaryEmbeddingJob.INDEX_SUMMARY_EMBEDDING_FIELD_PREFIX + "_" + embeddingModelId,
+            new String[] {
+                IndexSummaryEmbeddingJob.INDEX_NAME_FIELD,
+                IndexSummaryEmbeddingJob.INDEX_SUMMARY_FIELD,
+                IndexSummaryEmbeddingJob.INDEX_PATTERNS_FIELD,
+                IndexSummaryEmbeddingJob.DATA_STREAM_FIELD,
+                IndexSummaryEmbeddingJob.ALIAS_FIELD },
+            Optional.ofNullable(docSize).orElse(DEFAULT_K),
+            embeddingModelId,
+            Optional.ofNullable(k).orElse(DEFAULT_K)
+        );
+        this.mlClients = new MLClients(client);
+        this.inferenceModelId = inferenceModelId;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    protected Parser<SearchResponse, Object> searchResponseParser() {
+        return searchResponse -> {
+            SearchHit[] hits = searchResponse.getHits().getHits();
```

Comment:
Should it passes `doc_size` into the search?

Reply:
Here it is https://github.com/opensearch-project/skills/blob/main/src/main/java/org/opensearch/agent/tools/AbstractRetrieverTool.java#L89